tests/garage/tf/core/test_networks.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.networks import mlp
+from tests.fixtures import TfGraphTestCase
+
+
+class TestNetworks(TfGraphTestCase):
+    def setUp(self):
+        super(TestNetworks, self).setUp()
+        self.obs_input = np.array([[1, 2, 3, 4]])
+        input_shape = self.obs_input.shape[1]  # 4
+        self.hidden_nonlinearity = tf.nn.relu
+
+        self._input = tf.placeholder(
+            tf.float32, shape=(None, input_shape), name="input")
+
+        self._output_shape = 2
+
+        # We build a default mlp
+        self.mlp_f = mlp(
+            input_var=self._input,
+            output_dim=self._output_shape,
+            hidden_sizes=(32, 32),
+            hidden_nonlinearity=self.hidden_nonlinearity,
+            name="mlp1")
+
+        self.sess.run(tf.global_variables_initializer())
+
+    def test_multiple_same_mlp(self):
+        # We create another mlp with the same name
+        self.mlp_same_copy = mlp(
+            input_var=self._input,
+            output_dim=self._output_shape,
+            hidden_sizes=(32, 32),
+            hidden_nonlinearity=self.hidden_nonlinearity,
+            name="mlp1")
```

Comment:
What's the purpose for this test? Is this the reason you use `with tf.variable_scope(name, reuse=tf.AUTO_REUSE)` in mlp function?

Reply:
I want to show how to reuse the same mlp by referring to the same network name.